Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageLoader/QtImageLoader.cpp

```diff
@@ -22,14 +23,90 @@ namespace ImageProcessingAtom
 {
     namespace QtImageLoader
     {
+        /*
+        * https://www.opennet.ru/docs/formats/targa.pdf
+        * http://www.paulbourke.net/dataformats/tga/
+        */
+        enum class ImageTypeCode : AZ::u8
+        {
+            // No image data included.
+            NoImageData = 0, // not supported?
+            // Uncompressed, color-mapped images.
+            ColorMappedUncompressed = 1, // not supported?
+            // Uncompressed, RGB images.
+            UnmappedRGB = 2,
+            // Uncompressed, black and white images.
+            BlackAndWhiteUncompressed = 3, // not supported?
+            // Runlength encoded color-mapped images.
+            ColorMappedRLE = 9, // not supported?
+            // Runlength encoded RGB images.
+            RGBRLE = 10,
+            // Compressed, black and white images.
+            BlackAndWhiteCompressed = 11, // not supported?
+            // Compressed color-mapped data, using Huffman, Delta, and runlength encoding.
+            ColorMappedCompressedRLE = 32, // not supported?
+            // Compressed color-mapped data, using Huffman, Delta, and runlength encoding.4 - pass quadtree - type process.
+            ColorMappedCompressedRLEWith4Pass = 33, // not supported?
+        };
+        /*
+         * https://www.opennet.ru/docs/formats/targa.pdf
+         * http://www.paulbourke.net/dataformats/tga/
+         */
+        enum class ImagePixelSize : AZ::u8
+        {
+            Targa8   = 8,
+            Targa16 = 16,
+            Targa24 = 24,
+            Targa32 = 32,
+        };
+#pragma pack(1)
```

Comment:
Prefer using C++ bitfields to `#pragma pack`. The issue with `#pragma pack` is that it changes the natural alignment of the struct itself. So in this case, an instance of `TgaHeader` could reside at address 3 (instead of natural boundaries that are more efficient to load and store.

Reply:
This structure is to matching the actual tga file header size which is 18bytes. the c++ bitfields won't work well unless we manually converting each value after reading to raw buffer